File: app/Model/OEmbedable.php
<?php
class OEmbedable extends AppModel
{
    public function toOEmbed($data)
    {                
        $serverUrl = $_SERVER['SERVER_NAME'];
        $slug = $this->getData($this->alias.".slug");
        $iframeUrl = sprintf("http://%s/%ss/embed/%s/", $serverUrl, strtolower($this->alias), $slug);
        $url = sprintf("http://%s/%ss/view/%s/", $serverUrl, strtolower($this->alias), $slug);
        return array(          
            "version"   => "1.0",
            "type"      => "rich",
            "provider_name" => "The Music Tank",
            "provider_url"  => sprintf("http://%s/", $serverUrl),
            "url"       => $url,
            "title"     => $this->getData($this->alias.".title"),
            "data"      => $data,
            "width"     => 500,
            "height"    => 350,
            "html"      => '<iframe width="500" height="350" src="'.$iframeUrl.'" frameborder="0"></iframe>'
        );
    }
    
    public function getOEmbedUrl()
    {
        $serverUrl = $_SERVER['SERVER_NAME'];
        $destination = sprintf("http://%s/%ss/view/%s/", $serverUrl, strtolower($this->alias), $this->getData($this->alias.".slug"));
        return sprintf("http://%s/oembed?url=%s", $serverUrl, urlencode($destination));
    }
}

File: app/models/entity/oembedable.rb
module Entity
    # Allows TMT objects to be included in social media
    module Oembedable
        def to_oembed_hash
            hash = {
                "version"       => "1.0",
                "type"          => "rich",
                "provider_name" => "The Music Tank",
                "provider_url"  => "http://www.themusictank.com/",
                "width"         => 1280,
                "height"        => 720,
                "title"         => self.meta_title.join(", "),
                # "author_name": "",
                # "author_url": "http://www.themusictank.com/profiles/francoisfaubert",
                "html"          => "<iframe width=\"1280\" height=\"720\" src=\"" + link_to_me + "\"></iframe>",
            }
        end

        private

        def link_to_me
            "http://www.themusictank.com/embed/#{self.class.name.demodulize.downcase}s/#{self.slug}"
        end

    end
end
